concrete/src/Validator/ClosureValidator.php

```diff
@@ -1,56 +1,71 @@
 <?php
+
 namespace Concrete\Core\Validator;
 
+use ArrayAccess;
+use Closure;
+
+/**
+ * A generic validator cabable of describing itself and validating mixed values using closures.
+ */
 class ClosureValidator implements ValidatorInterface
 {
-    /** @var \Closure */
+    /**
+     * The closure that handles validation.
+     *
+     * @var \Closure
+     */
     protected $validator_closure;
 
-    /** @var \Closure */
+    /**
+     * The closure that returns requirements.
+     *
+     *@var \Closure
+     */
     protected $requirements_closure;
 
     /**
      * ClosureValidator constructor.
      *
-     * @param \Closure $validator_closure function(ClosureValidator $validator,
-     *                                             mixed $passed,
-     *                                             \Concrete\Core\Error $error = null): bool
+     * @param \Closure $validator_closure function(ClosureValidator $validator, mixed $passed, \Concrete\Core\Error $error = null): bool
      * @param \Closure $requirements_closure function(ClosureValidator $validator): array
+     *
+     * @example
      */
-    public function __construct(\Closure $validator_closure, \Closure $requirements_closure)
+    public function __construct(Closure $validator_closure, Closure $requirements_closure)
```

Comment:
Not something to change, just a note in the future we should be using `callable` rather than hinting on `\Closure`